custom_components/dualmode_generic/climate.py

```diff
@@ -302,13 +386,30 @@ def hvac_action(self):
             return CURRENT_HVAC_COOL
         if self._hvac_mode == HVAC_MODE_HEAT:
             return CURRENT_HVAC_HEAT
+        if self._hvac_mode == HVAC_MODE_HEAT_COOL:
+            if self.hass.states.is_state(self.heater_entity_id, STATE_ON):
```

Comment:
Yeah you're right, I initially tried to come up with a way of working this out without checking the switches but gave up. It's probably possible, I might take another stab at it. Will add a note to the readme about it otherwise, should get to look at it tonight.

Reply:
Ok, I had another look at it and given that it's used a couple of times I think a more significant refactor to support some sort of internal state would really be needed to fix the heater/cooler switch assumption.
So I added the note to the readme along with a couple of notes documenting the new features.
I decided to leave out the humidity sensor for now since it's unused.
Let me know if all good, or I can take another shot at it 👍